File: php/66-plus-one.php
<?php
class Solution {

    /**
     * @param Integer[] $digits
     * @return Integer[]
     */
    function plusOne($digits) {
        $carry = 0;
        $count = count($digits);
        for ($i = $count - 1; $i >= 0; $i--) {
            if ($i == $count - 1) {
                $newNum = $digits[$i] + 1;
            } else {
                $newNum = $digits[$i];
            }
			$newNum += $carry;
            $carry = intval($newNum / 10);
            $num = $newNum % 10;
            $digits[$i] = $num;
        }
        if ($carry) {
            array_unshift($digits, $carry);
        }
        return $digits;
    }
}

//test
$s = new Solution();
var_dump($s->plusOne([9, 9]));

File: go/66-plus-one.go
package main

func plusOne(digits []int) []int {
	next := 0
	res := make([]int, len(digits))
	for i := len(digits) - 1; i >= 0; i-- {
		val := 0
		if i == len(digits)-1 {
			val += 1
		}
		val += digits[i] + next
		next = val / 10
		res[i] = val % 10
	}
	if next != 0 {
		return append([]int{next}, res...)
	}
	return res
}
